fibers/conditions.scm

```diff
@@ -39,17 +39,42 @@
             wait-operation
             wait))
 
+
+;;; Counter utilities
+;;;
+;;; Counters here are an atomic box containing an integer which are
+;;; either incremented or reset.
+
+(define (counter-increment! counter)
```

Comment:
Can we add a `make-counter` above?  Of course it just does a `(make-atomic-box 0)`.